### PR TITLE
Add rule to check for existing classes, type aliases and shadowing of generic callables in properties.

### PR DESCRIPTION
**Pull request by mad-briller**
another follow up to: https://github.com/phpstan/phpstan-src/pull/2938#pullrequestreview-1899509254

covers native properties

### CODE REVIEW

**Comment by mad-briller** [comment]
is there a way to disable the `Lint` action for a file on a specific php version or do i have to exclude it for all / make the test file compatible with all?

File: tests/PHPStan/Rules/PhpDoc/data/generic-callables-incompatible.php [hunk 1]
```diff
@@ -191,3 +191,14 @@ function shadowsParamOutArray(array &$existingClasses): void
 function shadowsReturnArray(): array
 {
 }
+
```
Comment (ondrejmirtes) [inline tests/PHPStan/Rules/PhpDoc/data/generic-callables-incompatible.php:194]: Do `// lint >= 7.4` at the top of the file, see many comments like that in tests files :)

Comment (mad-briller) [reply tests/PHPStan/Rules/PhpDoc/data/generic-callables-incompatible.php:194]: good to know, dunno how i never noticed those before

**Comment by ondrejmirtes** [comment]
Thank you!

**Comment by mad-briller** [comment]
Currently without a dev env to continue on with the other steps, I'll try to follow up when I get one sorted

**Pull request merged**